### DISCUSSION TITLE
Custom Activity Editor, Conditional Properties

### DISCUSSION THREAD

**Discussion by mickdelaney** [elsa-workflows/elsa-core#641] [Q&A] [answered]
Just started looking into Elsa, seems a really great piece of work, so well done @sfmskywalker and all the other contributors !!
I asked this on Gitter but wasnt sure if this was a better place to ask.

One question I had was related to Custom Activities, and they're Properties.

I'm working through an email marketing workflow, and i have a Send Email Activity for it, but it has an Option for asking when it would be sent options could be:

- 'Immediately' 
- 'At a relative time period' 
- 'At a specific date/time'

If they select 'At a relative time period' then i'd like to display more properties. e.g Amount (integer) & Type (Select),
e.g. amount: 1, type: Day

I'm assuming as of V1x this is not something that the default Activity bits could cover, ActivityProperty, etc. 
And that I cant add something custom ?

Also part of this is that they'd need to Select an Email from a List, so an AutoComplete or dynamic Select.

One work around I've been looking at is to have 3 different Activities (Send Email Immedately, Send Email At a Relative Time .. ) etc. And Just use a text input for the email name. 

Thanks for your help ;-)

**Comment by sfmskywalker** [comment, marked as answer]
Hi Mick,

I've seen your question on Gitter, but this new Discussions feature from GitHub is another great way to ask your question (and presumably easier to find for others as well!)

I'll post my reply here as well for completeness' sake.

So, if you're doing this with Elsa 1, you could achieve what you are looking to do, but you'd have to implement a custom property field driver in StencilJS, which would allow you to provide any sort of UI you'd like.
Unfortunately I have none of that documented (yet).

For Elsa 2, it will be a similar story (the designer component for V2 is also built with StencilJS), but since it's a rewrite, its documentation will be different and more complete.

This doesn't help you right now of course, so your suggestion to create 3 different activities for now seems like a good workaround for the time being, as an alternative to cloning the StencilJS project and hardcoding in your custom fields.

**Comment by mickdelaney** [comment]
Thanks spike, I like stencil, have used it before. And I’m working on the elsa2x bits so eventually I’ll definitely look at the custom component. For now I’m actually defining an editor for the workflow, it’s a specific enough workflow (segmented email marketing automation) that I think it should get the job done initially.

**Comment by work-flower** [comment]
Hello guys,

Do you have any tutorial for this or is it enough just to check StencilJS?

Thanks...

**Reply by work-flower** [reply]
@sfmskywalker @mickdelaney thanks
